server/game/cards/09-HoT/FacelessMan.js

```diff
@@ -0,0 +1,34 @@
+const DrawCard = require('../../drawcard.js');
+
+class FacelessMan extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Gain character\'s icons, keywords, factions and traits',
+            limit: ability.limit.perPhase(1),
+            phase: 'challenge',
+            target: {
+                cardCondition: card => card.location === 'dead pile' && card.getType() === 'character'
+            },
+            handler: context => {
+                let copyObj = {
+                    icons: context.target.getIcons(),
+                    keywords: context.target.getPrintedKeywords(),
+                    factions: context.target.getFactions(),
+                    traits: context.target.getTraits()
+                };
+
+                this.untilEndOfPhase(ability => ({
+                    match: this,
+                    effect: ability.effects.gainIconsKeywordsFactionsTraits(copyObj)
```

Comment:
Although concise, this is super specific to this card.  Would it not be better to have separate effects for each 'part' gained? @ystros ?

Reply:
All options are bad here.
1. (Current) Introducing an effect that will likely only ever be used for one card - this one - is annoying.
2. You could add individual effects that will add an array of factions, array of traits, etc, but again, that feels somewhat one-off.
3. You could modify the `addTrait`, `addFaction`, etc effects to take either a value or an array. But doing so could make those effects less clear / implementation would have to be careful otherwise it would introduce bugs.
4. You could build up this effect using existing effects and bundling it into an array either using a bunch of concats or destructuring. Syntax gets nasty though.
```javascript
this.untilEndOfPhase(ability => ({
  match: this,
  effect: [
    ...context.target.getIcons().map(icon => ability.effects.addIcon(icon),
    ...context.target.getTraits().map(trait => ability.effects.addTrait(trait),
    // etc
  ]
});
```

Reply:
I had considered options 1 to 3, but 4 is probably the most straightforward solution and it has the advantage of keeping all annoying card-specific code on the card itself. So I went with that.